api/kroxylicious-filter-api/src/main/java/io/kroxylicious/proxy/filter/SafeInvoker.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.filter;
+
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.message.ResponseHeaderData;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ApiMessage;
+
+record SafeInvoker(FilterInvoker invoker) implements FilterInvoker {
```

Comment:
SafeInvoker is worthy of a javadoc

Reply:
Added docs, I think it's worth a follow up PR since it's raised questions here and in the initial PR adding it. We decided that it was a better interface if the `FilterHandler` could call `onRequest` or `onResponse` and have the invoker decide what to do with it even if it isn't interested in that particular decoded message. But I see now we've got a mismatch in the FilterInvoker javadoc where it says:

```
Handle deserialized request data. It is implicit that the underlying filter wants to handle this data
...
default void onRequest(...)
```

We should update the javadocs to say that `onRequest` and `onResponse` can be called for any decoded message and the filter should handle forwarding if it's not interested in that particular decoded message. 

This means we'd need to make all FilterInvoker implementations have this property and the SafeInvoker should disappear.